data-loader/core/src/main/java/com/scalar/db/dataloader/core/util/TableMetadataUtil.java

```diff
@@ -0,0 +1,144 @@
+package com.scalar.db.dataloader.core.util;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.dataloader.core.Constants;
+import com.scalar.db.dataloader.core.dataimport.controlfile.ControlFileTable;
+import com.scalar.db.io.DataType;
+import com.scalar.db.transaction.consensuscommit.Attribute;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/** Utility class for handling ScalarDB table metadata operations. */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TableMetadataUtil {
+
+  /**
+   * Determines whether a given column is a metadata column based on predefined criteria.
+   *
+   * @param columnName The name of the table column to check.
+   * @param metadataColumns A set of predefined metadata columns.
+   * @param columnNames A set of all column names in the table.
+   * @return {@code true} if the column is a metadata column; {@code false} otherwise.
+   */
+  public static boolean isMetadataColumn(
+      String columnName, Set<String> metadataColumns, Set<String> columnNames) {
+    if (metadataColumns.contains(columnName)) {
+      return true;
+    }
+    return columnName.startsWith(Attribute.BEFORE_PREFIX)
+        && !columnNames.contains(Attribute.BEFORE_PREFIX + columnName);
+  }
+
+  /**
+   * Determines whether a given column is a metadata column using table metadata.
+   *
+   * @param columnName The name of the ScalarDB table column to check.
+   * @param tableMetadata The metadata of the table.
+   * @return {@code true} if the column is a metadata column; {@code false} otherwise.
+   */
+  public static boolean isMetadataColumn(String columnName, TableMetadata tableMetadata) {
+    Set<String> metadataColumns = getMetadataColumns();
+    LinkedHashSet<String> columnNames = tableMetadata.getColumnNames();
+    return isMetadataColumn(columnName, metadataColumns, columnNames);
+  }
+
+  /**
+   * Retrieves a set of fixed metadata column names used in ScalarDB.
+   *
+   * @return A set of predefined metadata column names.
+   */
+  public static Set<String> getMetadataColumns() {
```

Comment:
How about using `com.scalar.db.transaction.consensuscommit.ConsensusCommitUtils#getTransactionMetaColumns` to avoid duplicated logic?

Some other methods might be replaced with `ConsensusCommitUtils` methods.